validator/client/attest_protect.go

```diff
@@ -97,6 +97,20 @@ func (v *validator) postAttSignUpdate(ctx context.Context, indexedAtt *ethpb.Ind
 		v.attesterHistoryByPubKey[pubKey] = newHistory
 	} else {
 		log.WithField("publicKey", fmtKey).Debug("Could not get local slashing protection data for validator in post validation")
+		history := kv.NewAttestationHistoryArray(indexedAtt.Data.Target.Epoch)
+		newHistory, err := kv.MarkAllAsAttestedSinceLatestWrittenEpoch(
+			ctx,
+			history,
+			indexedAtt.Data.Target.Epoch,
+			&kv.HistoryData{
+				Source:      indexedAtt.Data.Source.Epoch,
+				SigningRoot: signingRoot[:],
+			},
+		)
+		if err != nil {
+			return errors.Wrapf(err, "could not mark epoch %d as attested", indexedAtt.Data.Target.Epoch)
+		}
+		v.attesterHistoryByPubKey[pubKey] = newHistory
```

Comment:
This is duplicated on lines 85-97. Recommend: move to after the if/else block to deduplicate the code